### ISSUE TITLE
`yield from` seems to yield incorrect type analysis

### ISSUE THREAD

**Issue by rwols** [microsoft/pyright#897]
**Describe the bug**
When doing a `yield from` a `List[T]` in a function with return type `Generator[T, None, None]`, then everything should be OK. Instead pyright complains about incompatible return types.

**To Reproduce**
Use the following file:
```python
from typing import Generator, List

ints1 = [1, 2]
ints2 = [3, 4]


def foo() -> Generator[int, None, None]:
    yield from ints1
    yield from ints2


if __name__ == "__main__":
    print(list(foo()))
```
Then run pyright.

**Expected behavior**
No reported errors. Apparently one can yield from a list.

**Screenshots or Code**
N/A

**VS Code extension or command-line**
```
~/d/pyright-tests $ ls
foo.py
~/d/pyright-tests $ python3 foo.py 
[1, 2, 3, 4]
~/d/pyright-tests $ mypy .
Success: no issues found in 1 source file
~/d/pyright-tests $ pyright
No configuration file found.
stubPath /home/raoul/Documents/Programming/pyright-tests/typings is not a valid directory.
Searching for source files
Found 1 source file
/home/raoul/Documents/Programming/pyright-tests/foo.py
  8:16 - error: Expression of type "list[int]" cannot be assigned to yield type "Iterator[int]"
  "__next__" is not present (reportGeneralTypeIssues)
  9:16 - error: Expression of type "list[int]" cannot be assigned to yield type "Iterator[int]"
  "__next__" is not present (reportGeneralTypeIssues)
2 errors, 0 warnings 
Completed in 0.512sec
```

**Additional context**
N/A

**Comment by erictraut** [comment]
Thanks for the bug report. Internally, Pyright was using "Iterator" in a place where it should have been using "Iterable". This is what caused the incorrect error to be reported. This will be fixed in the next version of Pyright and Pylance.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.59, which I just published. It will also be included in the next release of Pylance.
